profile.go

```diff
@@ -84,6 +88,19 @@ func MemProfileRate(rate int) func(*Profile) {
 	}
 }
 
+// MemProfileType changes which type of memory profiling to
+// enable. The only two memory types supported are alloc and heap. It
+// disables any previous profiling settings.
+func MemProfileType(memProfileType string) func(*Profile) {
```

Comment:
I'd prefer if this was two func ops, perhaps `MemProfileHeap` and `MemProfileAllocs`